### ISSUE TITLE
can't configure an interface which is not yet available in /etc/network/interfaces

### ISSUE THREAD

**Issue by mjrider** [inkblot/puppet-network#1]
i tried to configure eth2, which is not yet configured in /etc/network/interfaces

interfaces:
auto lo
iface lo inet loopback

allow-hotplug eth0
iface eth0 inet dhcp

with puppet resource edit and puppet apply i tried to set the following confguration
network_interface { 'eth2':
  ensure  => 'present',
  hotplug => 'true',
  method  => 'static',
  address => '192.168.1.2/24', 
  onboot  => 'true',
}

which results in the following error
Error: /Stage[main]/Main/Network_interface[eth2]: Could not evaluate: undefined method `map' for nil:NilClass

if i change eth2 to eth0, or add an dummy configuration for eth2 in the interfaces file it works as expected
full log is available on:
https://gist.github.com/mjrider/c6955d1f56fc0bbea95f

**Comment by mjrider** [comment]
did some more debugging
in the method to_config, line 148 in lib/puppet/provider/network_interface/ifupdown.rb
runs a map on options which is nil

**Comment by inkblot** [comment]
Sorry I haven't had a chance to look at this until now. I'm pretty sure I tested the new interface scenario, but I'm seeing this error now too and I'm trying to understand how `options` got to be nil in the first place.

**Comment by inkblot** [comment]
Specifying `options => {}` in the manifest works around the issue, but this is unsatisfying.

**Comment by inkblot** [comment]
I've actually found another issue. This interface definition in the `interfaces` file breaks the resource's prefetch:

```
iface dummy0 inet static
    address 192.168.42.42
    netmask 255.255.255.0
```

**Comment by inkblot** [comment]
The prefetch break was due to some improper usage of Array.sort_by. There is a fix for that in 2.1.1 in addition to #2 .
